### ISSUE TITLE
InhabitType 4 (Flying)

### ISSUE THREAD

**Issue by ghost** [TrinityCore/TrinityCore#2858]
Creatures with only InhabitType=4 seems to respawn under the map (or fall) after you have killed them.

so you can only kill/target them once a serverstart.

it's annoying when this creature is unique and you require this for some quest - so you have to wait for a server down.
##

**Comment by Killyana** [comment]
This issue is not fixed, and it's annoying in some cases.

**Comment by Aokromes** [comment]
This bug is still valid after guid rewrite @r00ty-tc ?

**Comment by tkrokli** [comment]
I would like to test this, I just need some way to filter out non-friendly and attackable creatures.
I looked up `InhabitType`=4 in TDB 335.61 + updates up to and including `2016_06_14_05_world.sql`
```
SELECT * FROM `creature_template` WHERE `InhabitType`= 4;
```
but I get 1073 rows, many of them triggers, bunnies, friendly and NPCs with gossip. Which filters should I apply to find those I can fight and see if they fall into the ground?

**Comment by Killyana** [comment]
This still happen, find any flying that doesn't move, kill it and respawn it.

**Comment by r00ty-tc** [comment]
> This still happen, find any flying npc that doesn't move, kill it and respawn it.

If you know of one (presumably you do, knowing it's still a current issue), hit the dbguid or entry here so we can easily find it.

**Comment by Killyana** [comment]
This one : DB GUID: 152165,  Entry: 29747.

**Comment by r00ty-tc** [comment]
OK thanks. I'll try to take a look a little later.

**Comment by r00ty-tc** [comment]
OK first point, it doesn't happen on dynamic_spawning branch, for obvious reasons (it makes a brand new spawn and leaves the corpse behind when you respawn).

Also agreed it's client side (at least partially). Indeed if you attack them move to the original position, your attacks will land, and then it will rise up to position... Which all looks weird and funny.

**Comment by r00ty-tc** [comment]
OK. So the end result is this.

The cause, so far as I can see is that the spline for falling is never removed (server side or client side). So, when respawn happens, the spline is still active. It's propagated to players and they see the thing fall.

Now, there's a somewhat hacky way to solve this. Just create a new spline for the creature when you set state to DEAD.

If you as a player do .respawn while the corpse is still there it won't work. Simply because:

1: the creature is relocated. Which, if the full cycle of destroy/create isn't actioned, the location client side will be wrong.
2: For the same reason also the spline won't be cleared client side, even if you use near teleport.

In short, the semi hacky solution will fix it for normal operation, but not for .respawn. Dynamic Spawning branch solves the problem in all aspects, mainly because this problem only exists because of the current non-standard way respawns are done in core.
